AdblockPlusSafari/CustomFilterListsController.m

```diff
@@ -0,0 +1,297 @@
+    FilterList *filterList = [[FilterList alloc] initWithDictionary:self.adblockPlus.filterLists[CustomFilterListName]];
+    headerFooterView.textLabel.attributedText = [self footerTitleForFilterList:filterList];
+  }
+}
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+  UITableViewCell *cell = [tableView cellForRowAtIndexPath:indexPath];
+
+  if ([cell.reuseIdentifier isEqualToString:@"UpdateFilterLists"]) {
+    [self.adblockPlus updateActiveFilterLists:YES];
+    [self.tableView deselectRowAtIndexPath:indexPath animated:YES];
+  }
+}
+
+#pragma mark - UITextField
+
+- (BOOL)textFieldShouldReturn:(UITextField *)textField
+{
+  self.customFilterListUrl = textField.text;
+  [self createOrUpdateFilterListFromText:textField.text];
+  [textField resignFirstResponder];
+  return NO;
+}
+
+#pragma mark - Private
+
+- (NSURL *)createFilterListURLFromText:(NSString *)input
+{
+  // Trim input
+  NSString *url = [input stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
+
+  if ([url length] == 0) {
+    return nil;
+  }
+
+  // Prepend scheme if needed
+  if (![url hasPrefix:@"http://"] && ![url hasPrefix:@"https://"]) {
+    url = [@"http://" stringByAppendingString:url];
+  }
+
+  return [NSURL URLWithString:url];
+}
+
+- (void)createOrUpdateFilterListFromText:(NSString *)text
+{
+  NSURL *url = [self createFilterListURLFromText:text];
+  if (!url) {
+    return;
+  }
+
+  FilterList *customFilterList = [[FilterList alloc] initWithDictionary:@{}];
+  customFilterList.fileName = customFilterListFileName;
+  customFilterList.url = url.absoluteString;
+  NSMutableDictionary<NSString *, id> *filterLists = [self.adblockPlus.filterLists mutableCopy];
+  filterLists[CustomFilterListName] = customFilterList.dictionary;
+  self.adblockPlus.filterLists = filterLists;
+
+  [self.adblockPlus updateFilterListsWithNames:@[CustomFilterListName] userTriggered:NO];
+}
+
+- (NSDateFormatter *)dateFormatter
+{
+  NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+  dateFormatter.locale = [NSLocale currentLocale];
+  dateFormatter.dateStyle = NSDateFormatterFullStyle;
+  dateFormatter.timeStyle = NSDateFormatterShortStyle;
+  return dateFormatter;
+}
+
+- (NSString *)customFilterListUrl
+{
+  NSString *url = [NSUserDefaults.standardUserDefaults stringForKey:customFilterListKey];
+  if ([url length] == 0) {
+    return customFilterListUrl;
+  } else {
+    return url;
+  }
+}
+
+- (void)setCustomFilterListUrl:(NSString *)customFilterListUrl
+{
+  if ([customFilterListUrl length] > 0) {
+    [NSUserDefaults.standardUserDefaults setObject:customFilterListUrl forKey:customFilterListKey];
+  } else {
+    [NSUserDefaults.standardUserDefaults removeObjectForKey:customFilterListKey];
+  }
+  [NSUserDefaults.standardUserDefaults synchronize];
+}
+
+- (NSAttributedString *)footerTitleForFilterList:(FilterList *)filterList
+{
+  if (filterList) {
```

Comment:
This block has a few `else` after `return` - I think the code can be simplified a bit.